esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6047,20 +6034,54 @@ void CWsDfuEx::getReadAccess(IEspContext &context, IUserDescriptor *udesc, IEspD
 
     const char *keyPairName = getFileDafilesrvKeyName(*df);
 
+    // setup "expiryTime"
+    unsigned expirySecs = req.getExpirySeconds();
+
+    // JCSMORE - make this configurable
+    if (expirySecs > defaultMaxExpirySecs)
+        expirySecs = defaultMaxExpirySecs;
+
+    time_t now;
+    time(&now);
+    CDateTime expiryDt;
+    expiryDt.set(now + expirySecs);
+    StringBuffer expiryTimeStr;
+    expiryDt.getString(expiryTimeStr);
+
     Owned<IPropertyTree> metaInfo = createPTree();
-    getFileMeta(*metaInfo, *df, udesc, keyPairName, req);
+    getFileMeta(*metaInfo, *df, udesc, expiryTimeStr, keyPairName, req);
     StringBuffer metaInfoJsonBlob;
     toJSON(metaInfo, metaInfoJsonBlob);
     resp.setMetaInfoBlob(metaInfoJsonBlob);
     resp.setKeyName(keyPairName);
+    resp.setExpiryTime(expiryTimeStr);
 }
 
-bool CWsDfuEx::onDFUReadAccess(IEspContext &context, IEspDFUReadAccessRequest &req, IEspDFUReadAccessResponse &resp)
+bool CWsDfuEx::onDFUFileAccess(IEspContext &context, IEspDFUFileAccessRequest &req, IEspDFUFileAccessResponse &resp)
 {
     try
     {
-        if (!context.validateFeatureAccess(FEATURE_URL, SecAccess_Read, false) || req.getAccessType() != CSecAccessType_Read)
-            throw MakeStringException(ECLWATCH_DFU_ACCESS_DENIED, "Failed to CreateAndPublish. Permission denied.");
+        // JCSMORE would be better perhaps to have a translator helper between CSecAccessType and SecAccessFlags...
+        switch (req.getAccessType())
+        {
+            case CSecAccessType_None:
+                throw MakeStringException(ECLWATCH_DFU_ACCESS_DENIED, "onDFUFileAccess - Permission denied.");
+                break;
+            case CSecAccessType_Access:
+                break;
+            case CSecAccessType_Read:
+                if (!context.validateFeatureAccess(FEATURE_URL, SecAccess_Read, false))
+                    throw MakeStringException(ECLWATCH_DFU_ACCESS_DENIED, "onDFUFileAccess - Permission denied.");
+                break;
+            case CSecAccessType_Write:
+                if (!context.validateFeatureAccess(FEATURE_URL, SecAccess_Write, false))
+                    throw MakeStringException(ECLWATCH_DFU_ACCESS_DENIED, "onDFUFileAccess - Permission denied.");
+                break;
+            case CSecAccessType_Full:
+                if (!context.validateFeatureAccess(FEATURE_URL, SecAccess_Full, false))
+                    throw MakeStringException(ECLWATCH_DFU_ACCESS_DENIED, "onDFUFileAccess - Permission denied.");
+                break;
```

Comment:
case default?

Reply:
going to change to use a function that ransforms to SecAccessFlags